packages/wallet-core/src/protocols/direct-funder.ts

```diff
@@ -0,0 +1,253 @@
+import * as _ from 'lodash';
+
+import {checkThat, isSimpleAllocation, unreachable} from '../utils';
+import {Message} from '../wire-protocol';
+import {BN} from '../bignumber';
+import {addHash, hashState, signState} from '../state-utils';
+import {Address, SignatureEntry, State, StateWithHash, Uint256} from '../types';
+
+type AddressedMessage = {recipient: string; message: Message};
+
+export type OpenChannelEvent =
+  | {type: 'MessageReceived'; message: Message}
+  | {type: 'FundingUpdated'; amount: Uint256; finalized: boolean};
+
+type SignedStateHash = {hash: string; signatures: SignatureEntry[]};
+
+export enum WaitingFor {
+  theirPreFundSetup = 'DirectFunder.theirPreFundSetup',
+  safeToDeposit = 'DirectFunder.safeToDeposit',
+  channelFunded = 'DirectFunder.channelFunded',
+  theirPostFundState = 'DirectFunder.theirPostFundSetup'
+}
+
+export type OpenChannelObjective = {
```

Comment:
nit: might be cleaner to move types to their own file.

Reply:
Agreed, I'd do that if we choose to adopt this approach!